functorch/experimental/_cond.py

```diff
@@ -108,10 +110,34 @@ def _unwrap_proxy(e):
 def cond_dense(pred, true_fn, false_fn, operands):
     mode = _get_current_dispatch_mode()
     assert (mode is None), "Mode should never be enabled for CPU key"
-    if pred:
-        return true_fn(*operands)
+    # Determine which fn to call based on pred.
+    fn = true_fn if pred else false_fn
+    if isinstance(fn, GraphModule):
+        # It is possible that fn was defined in the current scope (i.e. scope of this call)
+        # and as such, it captured variables of the current scope in their closures.
+        # We need to look up the values of these variables and set them in its closure environment.
+        # Python does this automatically for functions, but here, fn is a graph module.
+        # When compiling, we expect captured variables of fn to be in special attributes of the graph module.
+        # By convention, fn.closure_{i} refers to the ith local of the scope containing fn.
+        current_frameinfo = next(frameinfo for frameinfo in inspect.stack() if frameinfo.function == "forward")
```

Comment:
This looks like you accidentally defined dynamic scoping rather than lexical scoping